src/include/homestore/btree/detail/btree_mutate_impl.ipp

```diff
@@ -95,37 +92,22 @@ retry:
 
         // Directly get write lock for leaf, since its an insert.
         child_cur_lock = (child_node->is_leaf()) ? locktype_t::WRITE : locktype_t::READ;
-
-        // If the child and child_info link in the parent mismatch, we need to do btree repair, it might have
-        // encountered a crash in-between the split or merge and only partial commit happened.
-        if (is_split_needed(child_node, req) || is_repair_needed(child_node, child_info)) {
+        if (is_split_needed(child_node, req)) {
```

Comment:
New design doesn't need run time repair and all repairs will be done during recovery, so removed that piece of code.

Reply:
I thought we still needed but a different purpose. If a pair is needed it means there is a bug in recovery . So an assert is needed instead.